cpp/ql/lib/semmle/code/cpp/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -374,12 +390,15 @@ private module Cached {
   }
 
   /**
-   * Holds if `p` is the `i`th parameter of a viable dispatch target of `call`.
-   * The instance parameter is considered to have index `-1`.
+   * Holds if `p` is the parameter of a viable dispatch target of `call`,
+   * and `p` matches arguments at position `apos`.
    */
   pragma[nomagic]
-  private predicate viableParam(DataFlowCall call, int i, ParamNode p) {
-    p.isParameterOf(viableCallableExt(call), i)
+  private predicate viableParam(DataFlowCall call, ArgumentPosition apos, ParamNode p) {
```

Comment:
The same join-order rewrite applies here.

Reply:
Hmm, wonder how I missed that. Thanks.